Model/Api/CreateOrder.php

```diff
@@ -0,0 +1,314 @@
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2019 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Model\Api;
+
+use Bolt\Boltpay\Api\CreateOrderInterface;
+use Bolt\Boltpay\Exception\BoltException;
+use Bolt\Boltpay\Helper\Cart;
+use Magento\Framework\Exception\LocalizedException;
+use Bolt\Boltpay\Helper\Order as OrderHelper;
+use Bolt\Boltpay\Helper\Cart as CartHelper;
+use Bolt\Boltpay\Helper\Log as LogHelper;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Webapi\Rest\Request;
+use Bolt\Boltpay\Helper\Hook as HookHelper;
+use Bolt\Boltpay\Helper\Bugsnag;
+use Magento\Framework\Webapi\Rest\Response;
+use Bolt\Boltpay\Helper\Config as ConfigHelper;
+
+/**
+ * Class CreateOrder
+ * Web hook endpoint. Create the order.
+ *
+ * @package Bolt\Boltpay\Model\Api
+ */
+class CreateOrder implements CreateOrderInterface
+{
+    const E_BOLT_GENERAL_ERROR = 2001001;
+    const E_BOLT_ORDER_ALREADY_EXISTS = 2001002;
+    const E_BOLT_CART_HAS_EXPIRED = 2001003;
+    const E_BOLT_ITEM_PRICE_HAS_BEEN_UPDATED = 2001004;
+
+    /**
+     * @var HookHelper
+     */
+    private $hookHelper;
+
+    /**
+     * @var OrderHelper
+     */
+    private $orderHelper;
+
+    /**
+     * @var LogHelper
+     */
+    private $logHelper;
+
+    /**
+     * @var Request
+     */
+    private $request;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @var Response
+     */
+    private $response;
+
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+
+    /**
+     * @var CartHelper
+     */
+    private $cartHelper;
+
+    /**
+     * @param HookHelper   $hookHelper
+     * @param OrderHelper  $orderHelper
+     * @param CartHelper   $cartHelper
+     * @param LogHelper    $logHelper
+     * @param Request      $request
+     * @param Bugsnag      $bugsnag
+     * @param Response     $response
+     * @param ConfigHelper $configHelper
+     */
+    public function __construct(
+        HookHelper $hookHelper,
+        OrderHelper $orderHelper,
+        CartHelper $cartHelper,
+        LogHelper $logHelper,
+        Request $request,
+        Bugsnag $bugsnag,
+        Response $response,
+        ConfigHelper $configHelper
+    ) {
+        $this->hookHelper = $hookHelper;
+        $this->orderHelper = $orderHelper;
+        $this->logHelper = $logHelper;
+        $this->request = $request;
+        $this->bugsnag = $bugsnag;
+        $this->response = $response;
+        $this->configHelper = $configHelper;
+        $this->cartHelper = $cartHelper;
+    }
+
+    /**
+     * Pre-Auth hook: Create order.
+     *
+     * @api
+     *
+     * @param string $type - "order.create"
+     * @param array $order
```

Comment:
The name here is a little misleading.  You have to variables that you are using: `$order` and `$createOrderData`.  Consider switching the names and then the names in context makes very clear self-documenting sense.

Reply:
These params come from the request and also implement Interface that should be equal to the request fields.

The request from Bolt: [{"type":"order.create","order":{},"currency":"USD"}]